File: typescript-connector/src/models/NutrientCharacteristic.ts
// Class from DFC Business Ontology: #NutrientCharacteristic
import { SemanticObject } from "../core/SemanticObject.js";

export interface NutrientCharacteristicParams {
  hasNutrientDimension?: string;
  nutrientCharacteristicOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
}
export class NutrientCharacteristic extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:NutrientCharacteristic";
  }

  hasNutrientDimension?: string;
  nutrientCharacteristicOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;

  constructor(
    semanticId: string,
    params?: NutrientCharacteristicParams,
  ) {
    super(semanticId);
    this.hasNutrientDimension = params?.hasNutrientDimension;
    this.nutrientCharacteristicOf = params?.nutrientCharacteristicOf;
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.semanticType = NutrientCharacteristic.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:has_nutrient_dimension", () => this.hasNutrientDimension);
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:nutrient_characteristic_of", () => this.nutrientCharacteristicOf);
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:date", () => this.date);
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:description", () => this.description);
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:name", () => this.name);
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:characteristic_of", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:NutrientCharacteristic:has_dimension", () => this.hasDimension);
  }
  static {
    SemanticObject.typeRegistry.set(NutrientCharacteristic.SEMANTIC_TYPE, NutrientCharacteristic);
  }
}

File: ruby-gem/lib/models/nutrient_characteristic.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #NutrientCharacteristic
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class NutrientCharacteristic < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:NutrientCharacteristic".freeze

      # @return [String]
      attr_accessor :nutrient_characteristic_of

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @return [Concept, String]
      attr_accessor :nutrient_dimension

      # @param semanticId [String]
      # @param nutrientCharacteristicOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, nutrientDimension: nil
      def initialize(semanticId, nutrientCharacteristicOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, nutrientDimension: nil)
        super(semanticId)
        @nutrient_characteristic_of = nutrientCharacteristicOf
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        @nutrient_dimension = nutrientDimension
        self.semanticType = "dfc-b:NutrientCharacteristic"
        registerSemanticProperty("dfc-b:nutrientCharacteristicOf", &method("nutrient_characteristic_of")).valueSetter = method("nutrient_characteristic_of=")
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
        registerSemanticProperty("dfc-b:hasNutrientDimension", &method("nutrient_dimension")).valueSetter = method("nutrient_dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
